packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts

```diff
@@ -84,79 +85,170 @@ export class ComponentUtils {
     }
   }
 
-  public async buildCustomObjectFieldsList(
-    objectName: string,
-    defaultUsernameOrAlias: string,
-    outputPath: string
-  ): Promise<string[]> {
-    const jsonResult = await forceListSchemaSobjectDescribe(
-      objectName,
-      defaultUsernameOrAlias,
-      outputPath
-    );
-
-    const result = JSON.parse(jsonResult);
-    const fields = result.result.fields.map(
-      (field: {
-        type: string;
-        relationshipName?: string;
-        name: string;
-        length?: number;
-      }) => {
-        switch (field.type) {
-          case 'string':
-          case 'textarea':
-          case 'email':
-            return `${field.name} (${field.type}(${field.length}))`;
-          case 'reference':
-            return `${field.relationshipName} (reference)`;
-          default:
-            return `${field.name} (${field.type})`;
-        }
+  public buildCustomObjectFieldsList(
+    sObject: string,
+    result?: string,
+    componentsPath?: string
+  ): string[] {
+    try {
+      if (isNullOrUndefined(result)) {
+        result = fs.readFileSync(componentsPath!, 'utf8');
       }
-    );
+      const jsonResult = JSON.parse(result);
+      const fields = jsonResult.result.map(
+        (field: {
+          name: string;
+          type: string;
+          relationshipName?: string;
+          length?: number;
+        }) => {
+          switch (field.type) {
+            case 'string':
+            case 'textarea':
+            case 'email':
+              return `${field.name} (${field.type}(${field.length}))`;
+            case 'reference':
+              return `${field.relationshipName} (reference)`;
+            default:
+              return `${field.name} (${field.type})`;
+          }
+        }
+      );
+      telemetryService.sendEventData('CustomObjects Fields quantity', { sObject }, { fields: fields.length });
+      return fields;
+    } catch (e) {
+      telemetryService.sendException('metadata_cmp_build_custom_objects_fields_list', e.message);
+      throw new Error(e);
+    }
+  }
+
+  public async listMetadataTypes(
+    metadataType: string,
+    connection: Connection,
+    componentsPath: string
+  ): Promise<string> {
+    const metadataQuery = {type: metadataType};
+    const metadataFileProperties = await connection.metadata.list(metadataQuery);
+    const result = {status: 0, result: metadataFileProperties};
+    const jsonResult = JSON.stringify(result);
+    fs.writeFileSync(componentsPath, jsonResult);
+    return jsonResult;
+  }
 
-    return fields;
+  public async listSObjectFields(
```

Comment:
Same comment as above. Consider to rename this method name.

Reply:
Renaming listSObjectFields to fetchAndSaveSObjectFieldsProperties.